types/tls.go

```diff
@@ -73,31 +57,19 @@ var (
 	}
 )
 
-func (v TLSVersion) String() string {
-	return ConsulConfigTLSVersionStrings[v]
-}
-
-func (v TLSVersion) MarshalJSON() ([]byte, error) {
-	return json.Marshal(v.String())
+func (v *TLSVersion) String() string {
+	return fmt.Sprintf("%s", *v)
 }
 
-func (v *TLSVersion) UnmarshalJSON(bytes []byte) error {
-	versionStr := string(bytes)
-
-	if n := len(versionStr); n > 1 && versionStr[0] == '"' && versionStr[n-1] == '"' {
-		versionStr = versionStr[1 : n-1] // trim surrounding quotes
-	}
-
-	if version, ok := TLSVersions[versionStr]; ok {
-		*v = version
-		return nil
+func ValidateTLSVersion(v TLSVersion) error {
+	if _, ok := tlsVersions[v.String()]; !ok {
```

Comment:
If the map were updated to use the TLSVersion as a key as above, then this wouldn't need the `v.String()` call